src/fcopy/main.c

```diff
@@ -100,6 +103,79 @@ static void selinux_relabel_path(const char *path, const char *inside_path) {
 #endif
 }
 
+// Convert string of comma separated values to NULL-terminated string array.
+// Original string is modified in-place.
```

Comment:
> // Original string is modified in-place.

It already returns the list, so why do that?

Why not e.g.: work on a copy of `clist` in the function?

Given the way that this is called on fcopy's `main()`:

>     keep_xattr = csv_to_strv(argv[++i]);

That appears to effectively result in modifying the argv item from something
like "name1,name2,name3" to point to the first item on the resulting list
(which would be "name1" in this case).

Allowing unprivileged users to turn arbitrary strings into globally-accessible
string arrays doesn't sound like a good idea to me.

Reply:
> It already returns the list, so why do that?
> 
> Why not e.g.: work on a copy of `clist` in the function?

To reduce unnecessary allocations and copying.

> Given the way that this is called on fcopy's `main()`:
> 
> > ```
> > keep_xattr = csv_to_strv(argv[++i]);
> > ```
> 
> That appears to effectively result in modifying the argv item from something like "name1,name2,name3" to point to the first item on the resulting list (which would be "name1" in this case).

It is. But it seems okay since this argument will not be used by anyone else after the call to `csv_to_strv`.

> Allowing unprivileged users to turn arbitrary strings into globally-accessible
string arrays doesn't sound like a good idea to me.

The only thing that I have on my mind is that `/proc/pid/cmdline` will not reflect the actual argv the process was called with. It doesn't seem like a big issue in a helper tool.